src/plugins/explore/public/application/register_tabs.ts

```diff
@@ -45,17 +47,31 @@ export const registerBuiltInTabs = (tabRegistry: TabRegistryService) => {
     supportedLanguages: [EXPLORE_DEFAULT_LANGUAGE],
 
     prepareQuery: (query) => {
-      return typeof query.query === 'string' ? query.query : '';
+      const state = services.store.getState();
+
+      // Get the selected patterns field from the Redux state
+      let patternsField = state.tab?.patterns?.patterns_field;
+
+      const preparedQuery = getQueryWithSource(query);
+      if (!patternsField) {
+        patternsField = findDefaultPatternsField(services);
+      }
+
+      return preparedQuery.query !== ''
+        ? preparedQuery.query +
+            ` | patterns \`${patternsField}\` method=brain mode=aggregation | sort - pattern_count`
```

Comment:
since you are already using string interperlation, lets just use that all the way. IE instead of:

```
? preparedQuery.query +
            ` | patterns \`${patternsField}\` method=brain 
```

lets do:

```
`${preparedQuery.query} | patterns \`${patternsField}\` method=brain mode=aggregation | sort - pattern_count`
```

Reply:
changed